app/2.3.x/external-plugins.md

```diff
@@ -0,0 +1,247 @@
+
+Plugins written in Lua define a schema to specify how to read and validate
+configuration data coming from the datastore or the Admin API.  Since Go is a
+statically type language, all that specification is handled by defining a
+configuration structure:
+
+```
+type MyConfig struct {
+    Path   string
+    Reopen bool
+}
+```
+
+Public fields (that is, those starting with a capital letter) will be filled
+with configuration data.  If you want them to have a different name in the
+datastore, add field tags as defined in the `encoding/json` package:
+
+```
+type MyConfig struct {
+    Path   string `json:my_file_path`
+    Reopen bool   `json:reopen`
+}
+```
+
+#### 2. New() Constructor
+
+Your plugin must define a function called New that creates an instance of this type
+and returns as an `interface{}`.  In most cases it’s just this:
+
+```
+func New() interface{} {
+    return &MyConfig{}
+}
+```
+
+You can add more fields to the structure and they’ll be passed around, but
+there’s no guarantees about the lifetime or quantity of configuration
+instances.
+
+#### 3. Phase Handlers
+
+Similarly to Kong Lua plugins, you can implement custom logic to be executed at
+various points of the request processing life-cycle. For example, to execute
+custom Go code in the "access" phase, define a function named `Access`:
+```
+func (conf *MyConfig) Access (kong *pdk.PDK) {
+  ...
+}
+```
+
+The phases you can implement custom logic for are, and the expected function
+signature is the same for all of them:
+- `Certificate`
+- `Rewrite`
+- `Access`
+- `Response`
+- `Preread`
+- `Log`
+
+Similar to Lua plugins, the presence of the `Response` handler automatically enables the "buffered proxy" mode.
+
+
+### Embedded server
+
+Each plugin can be a microservice, compiled as a standalone executable.
+
+To use the embedded server, include "github.com/Kong/go-pdk/server" in
+the imports list, and add a `main()` function:
+
+```go
+func main () {
+  server.StartServer(New, Version, Priority)
+}
+```
+
+Note that the `main()` function must have a `package main` line at the
+top of the file.
+
+Then, a standard go build creates an executable. no extra "go-pluginserver",
+no plugin loading, no compiler/library/environment compatibility issues.
+
+The resulting executable can be placed somewhere in your path (for example
+`/usr/local/bin`).  The common `-h` flag shows a "usage" help message:
+
+```
+$ my-plugin -h
+
+Usage of my-plugin:
+  -dump
+        Dump info about plugins
+  -help
+        Show usage info
+  -kong-prefix string
+        Kong prefix path (specified by the -p argument commonly used in the kong cli) (default "/usr/local/kong")
+```
+
+When run without arguments, it will create a socket file with the
+`kong-prefix` and the executable name, appending `.socket`.  For example,
+if the executable is `my-plugin` it would be
+`/usr/local/kong/my-plugin.socket` by default.
+
+Example configuration: two standalone plugins, called `my-plugin`
```

Comment:
Awkward to have a sentence with two colons. Suggest making a heading at the appropriate level.

```suggestion
## Example configuration

Two standalone plugins, called `my-plugin`
```

Reply:
done